pkg/workflow/controllers/abort_node_reconciler.go

```diff
@@ -0,0 +1,147 @@
+// Copyright Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/controllers/utils/recorder"
+)
+
+type AbortNodeReconciler struct {
+	*ChildNodesFetcher
+	kubeClient    client.Client
+	eventRecorder recorder.ChaosRecorder
+	logger        logr.Logger
+}
+
+func NewAbortNodeReconciler(kubeClient client.Client, eventRecorder recorder.ChaosRecorder, logger logr.Logger) *AbortNodeReconciler {
+	return &AbortNodeReconciler{
+		ChildNodesFetcher: NewChildNodesFetcher(kubeClient, logger),
+		kubeClient:        kubeClient,
+		eventRecorder:     eventRecorder,
+		logger:            logger}
+}
+
+// Reconcile watches `WorkflowNodes`, if:
+// 1. the abort condition is `False`, just return.
+// 2. the abort condition is `True`, the node is not `TypeStatusCheck`, it will propagate abort condition to children nodes.
+// 3. the abort condition is `True`, the node is `TypeStatusCheck`, it will add abort annotation to the parent workflow.
```

Comment:
No, the abort 'flow' is top-down, only the parent node will abort the children nodes.

Reply:
So I add another reconciler ([AbortWorkflowReconciler](https://github.com/iguoyr/chaos-mesh/blob/statuscheck-in-workflow/pkg/workflow/controllers/abort_workflow_reconciler.go#L49)), it will set the abort condition of the `entry node` to `True`.